app/src/main/java/com/nextcloud/talk/jobs/ContactAddressBookWorker.kt

```diff
@@ -212,147 +200,176 @@ class ContactAddressBookWorker(val context: Context, workerParameters: WorkerPar
-                        whereName,
-                        whereNameParams,
-                        ContactsContract.CommonDataKinds.StructuredName.GIVEN_NAME)
-                if (nameCursor != null) {
-                    while (nameCursor.moveToNext()) {
-                        displayName = nameCursor.getString(nameCursor.getColumnIndex(ContactsContract.CommonDataKinds.StructuredName.DISPLAY_NAME))
+                    if (displayName == null) {
+                        return
                     }
-                    nameCursor.close()
-                }
 
-                if (displayName == null) {
-                    return
-                }
+                    val ops = ArrayList<ContentProviderOperation>()
+                    val rawContactsUri = ContactsContract.RawContacts.CONTENT_URI.buildUpon().build()
+                    val dataUri = ContactsContract.Data.CONTENT_URI.buildUpon().build()
+
+                    ops.add(ContentProviderOperation
+                            .newInsert(rawContactsUri)
+                            .withValue(ContactsContract.RawContacts.ACCOUNT_NAME, accountName)
+                            .withValue(ContactsContract.RawContacts.ACCOUNT_TYPE, accountType)
+                            .withValue(ContactsContract.RawContacts.AGGREGATION_MODE,
+                                    ContactsContract.RawContacts.AGGREGATION_MODE_DEFAULT)
+                            .withValue(ContactsContract.RawContacts.SYNC2, cloudId)
+                            .build())
+                    ops.add(ContentProviderOperation
+                            .newInsert(dataUri)
+                            .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID, 0)
+                            .withValue(ContactsContract.Data.MIMETYPE, ContactsContract.CommonDataKinds.Phone.CONTENT_ITEM_TYPE)
+                            .withValue(ContactsContract.CommonDataKinds.Phone.NUMBER, numbers[0])
+                            .build())
+                    ops.add(ContentProviderOperation
+                            .newInsert(dataUri)
+                            .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID, 0)
+                            .withValue(ContactsContract.Data.MIMETYPE, ContactsContract.CommonDataKinds.StructuredName.CONTENT_ITEM_TYPE)
+                            .withValue(ContactsContract.CommonDataKinds.StructuredName.DISPLAY_NAME, displayName)
+                            .build())
+                    ops.add(ContentProviderOperation
+                            .newInsert(dataUri)
+                            .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID, 0)
+                            .withValue(ContactsContract.Data.MIMETYPE, "vnd.android.cursor.item/vnd.com.nextcloud.talk2.chat")
+                            .withValue(ContactsContract.Data.DATA1, cloudId)
+                            .withValue(ContactsContract.Data.DATA2, "Chat via " + accountName)
```

Comment:
Not translated?

Reply:
i tried but had some strange problems when i modified strings.xml :D but now it worked..